build.rs

```diff
@@ -29,5 +30,69 @@ fn main() -> Result<(), Box<dyn Error>> {
     writeln!(f, "pub const USER_AGENT: &str = \"sentry-cli/{arch}\";")?;
     println!("cargo:rerun-if-changed=build.rs\n");
 
+    if platform == "darwin" {
+        println!("cargo:rerun-if-changed=native/swift/AssetCatalogParser");
+
+        env::set_current_dir("native/swift/AssetCatalogParser")
+            .expect("Failed to change to AssetCatalogParser directory");
+
+        let target_dir = "../../../target/swift-bridge";
+
+        let _ = std::fs::remove_dir_all(".build");
+        let _ = std::fs::remove_file(format!("{}/libswiftbridge.a", target_dir));
+
+        let status = Command::new("swift")
+            .args([
+                "build",
+                "-c",
+                "release",
+                "--triple",
+                &format!("{}-apple-macosx11", arch),
+            ])
+            .status()
+            .expect("Failed to compile SPM");
+
+        assert!(status.success(), "swift build failed");
+
+        let target_dir = "../../../target/swift-bridge";
+
+        std::fs::create_dir_all(target_dir)
+            .expect("Failed to create target/swift-bridge directory");
+
+        let status = Command::new("ar")
+            .args([
+                "crus",
+                &format!("{}/libswiftbridge.a", target_dir),
+                ".build/release/AssetCatalogParser.build/AssetCatalogReader.swift.o",
+                ".build/release/ObjcSupport.build/safeValueForKey.m.o",
+            ])
+            .status()
+            .expect("Failed to create static library");
+
+        assert!(status.success(), "ar failed");
+
+        env::set_current_dir("../../../").expect("Failed to change back to original directory");
```

Comment:
if this is required for the Command, you can set the cwd on the Command instance itself. also i think in build scripts you are supposed to only write to OUT_DIR